src/Cli/dotnet/NugetPackageDownloader/NuGetPackageDownloader.cs

```diff
@@ -464,16 +521,27 @@ private IEnumerable<PackageSource> LoadNuGetSources(PackageSourceLocation packag
 
             _ = source ?? throw new ArgumentNullException(nameof(source));
 
-            SourceRepository repository = Repository.Factory.GetCoreV3(source);
-            PackageMetadataResource resource = await repository
-                .GetResourceAsync<PackageMetadataResource>(cancellationToken).ConfigureAwait(false);
-            IEnumerable<IPackageSearchMetadata> foundPackages = await resource.GetMetadataAsync(
-                packageIdentifier,
-                includePrerelease,
-                false,
-                _cacheSettings,
-                _verboseLogger,
-                cancellationToken).ConfigureAwait(false);
+            IEnumerable<IPackageSearchMetadata> foundPackages;
+
+            try
+            {
+                SourceRepository repository = Repository.Factory.GetCoreV3(source);
+                PackageMetadataResource resource = await repository
+                    .GetResourceAsync<PackageMetadataResource>(cancellationToken).ConfigureAwait(false);
+
+                foundPackages = await resource.GetMetadataAsync(
+                    packageIdentifier,
+                    includePrerelease,
+                    false,
+                    _cacheSettings,
+                    _verboseLogger,
+                    cancellationToken).ConfigureAwait(false);
+            }
+            catch (FatalProtocolException e) when (_restoreActionConfig.IgnoreFailedSources)
```

Comment:
turn FatalProtocolException to warning, the same as nuget. Comments are in the record type